### PR TITLE
Correct [] handling

### PR DESCRIPTION
**Pull request by mfalesni**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit f8603c26285df97f0f698b7cd13dc0c43ff56a5c:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: artifactor/plugins/reporter.py [hunk 1]
```diff
@@ -248,7 +249,12 @@ def build_dict(self, path, container, contents):
         Build a hierarchical dictionary including information about the stats at each level
         and the duration.
         """
-        segs = path.split('/')
+
+        if isinstance(path, basestring):
+            segs = process_pytest_path(path)
+        else:
+            segs = path
```
Comment (psav) [inline artifactor/plugins/reporter.py]: Do we expect this to eve not be a path?

File: artifactor/plugins/reporter.py [hunk 2]
```diff
@@ -262,7 +268,7 @@ def build_dict(self, path, container, contents):
             if head not in container['_sub']:
                 container['_sub'][head] = deepcopy(_tests_tpl)
             # Call again to recurse down the tree.
-            self.build_dict('/'.join(end), container['_sub'][head], contents)
+            self.build_dict(end, container['_sub'][head], contents)
```
Comment (psav) [inline artifactor/plugins/reporter.py]: Will this always have the leading slash?

File: artifactor/plugins/reporter.py [hunk 3]
```diff
@@ -285,9 +291,12 @@ def build_li(self, lev):
...
                     bimdict[v['outcomes']['overall']], v['outcomes']['overall'].upper())
-                link = ('<a href="#{}">{} {} {} '
-                        '<span style="color:#888888"><em>[{}]</em></span></a>').format(
-                    v['name'], os.path.split(v['name'])[1], teststring, label, pretty_time)
+                proc_name = process_pytest_path(v['name'])[-1]
+                link = (
+                    '<a href="#{}">{} {} {} <span style="color:#888888"><em>[{}]</em></span></a>'
+                    .format(v['name'], proc_name, teststring, label, pretty_time))
+                # Do we really need the os.path.split (now process_pytest_path) here?
+                # For me it seems the name is always the leaf
```
Comment (mfalesni) [inline artifactor/plugins/reporter.py]: Why the os.path.split was here @psav ? Could it smuggle more than just the leaf location of the node?

File: artifactor/plugins/reporter.py (hunk 1 above)
Comment (mfalesni) [inline artifactor/plugins/reporter.py]: Well, the build_dict calls itself recursively, so when it once breaks the string to the path, we don't need to convert it in the subsequent calls, we just pass the resst of the list.

File: artifactor/plugins/reporter.py (hunk 2 above)
Comment (mfalesni) [inline artifactor/plugins/reporter.py]: The code I wrote does not care about the leading slashes, it strips them away as the first step.

File: artifactor/plugins/reporter.py (hunk 1 above)
Comment (psav) [inline artifactor/plugins/reporter.py]: Noted

File: artifactor/plugins/reporter.py (hunk 2 above)
Comment (psav) [inline artifactor/plugins/reporter.py]: OK, and this is fine for later on? I guess I was asking if build_dict required the leading slash, but I guess if you refactored stuff it probably doesn't any more.

File: artifactor/plugins/reporter.py (hunk 3 above)
Comment (psav) [inline artifactor/plugins/reporter.py]: I'll be honest, I'm not sure, it's been a very long time since I looked at that code.

Comment (mfalesni) [inline artifactor/plugins/reporter.py]: Well, as it is refactored now, it should have the same behaviour as os.path.split()[1], so this should be ok as it is. I was just wondering if that can be avoided in the future :)

File: artifactor/plugins/reporter.py (hunk 2 above)
Comment (mfalesni) [inline artifactor/plugins/reporter.py]: I let it print the path strings and I think there were no leading slashes.

**Pull request merged**